Repository/ContainerRepositoryFactory.php

```diff
@@ -30,7 +29,7 @@ public function __construct(ContainerInterface $container = null)
     {
         // When DoctrineBundle requires Symfony 3.3+, this can be removed
         // and the $container argument can become required.
-        if ($container === null && class_exists(ServiceLocatorTagPass::class)) {
+        if ($container === null) {
             throw new InvalidArgumentException(sprintf('The first argument of %s::__construct() is required on Symfony 3.3 or higher.', self::class));
```

Comment:
Shouldn't we just make it non-nullable in constructor, since it throws exception otherwise anyway?

Reply:
Unfortunately, that would change the behaviour from an exception to a fatal error, which I would consider a BC break. We'll do that for 2.0.